functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,230 @@ export const processNewAllotment = functions.database
+                    }
+
+
+                    ///////////////////////////////
+                    // Sorting by Beginning time
+                    ///////////////////////////////
+                    let chunks = lastFile.chunks;
+
+                    // `start` attribute is added to be used in sorting
+                    chunks.forEach(element => element.start = moment.duration(element.beginning).asSeconds());
+                    chunks = lodash.orderBy(chunks, ['start'], ['asc']);
+
+                    // getting rid of the added `start` attribute
+                    chunks.forEach(element => delete element.start);
+
+
+
+                    ////////////////////////////////////////////////////////
+                    // Filling `continuationFrom` only for the FIRST column
+                    ////////////////////////////////////////////////////////
+                    for (let z = 0; z < chunks.length; z++) 
+                        if (chunks[z].continuationFrom && z !== 0)
+                            delete chunks[z].continuationFrom;
+
+
+
+                    //////////////////////////////////////
+                    // Ensuring data is NEVER overwritten
+                    //////////////////////////////////////
+                    let ref = await db
+                        .ref(`/sound-editing/chunks/${sheetTitles[i].title}/${lastFile.file_name}`)
+                        .once('value');
+
+                    if (!ref.exists())
+                        await db.ref(`/sound-editing/chunks/${sheetTitles[i].title}/${lastFile.file_name}`)
+                            .set(lastFile.chunks);
+                    else {
+                        let data = ref.val();
+                        for (let i = 0; i < data.length; i++) {
+                            // Instead of comparing every single attribute, the two object are converted
+                            // into strings and then compared
+                            if (JSON.stringify(data[i]) != JSON.stringify(lastFile.chunks[i])) {
+                                summary.modified.push(lastFile);
+                                break;
+                            }                        
+                        }
+                    }                
+                }
+                
+
+                // 1. New chunk doesn't belong to the last audio file
+                //          --> last audio file chunks were saved to firebase 
+                //          --> Reset `lastFile` object to hold the last read row 
+                //                         OR
+                // 2. last chunk is `skipped` and therefor all of the previous chunks
+                //      of the same audioFile ( the ones in the `lastFile` object ) is discarded
+                //          --> Reset `lastFile` object to hold the last row
+
+
+                lastFile.file_name = row[columnsIndex['AudioFileName']];
+                lastFile.chunks = [{
```

Comment:
Here you're repeating the same code as above (line 204). This is an anti-pattern. Rewrite the code so that you don't copy-paste it.